### PR TITLE
Check::Encoding: use charlock_holmes instead of builtin ruby hacks

### PR DESCRIPTION
**Pull request by dunn**

### CODE REVIEW

File: lib/check/encoding.rb [hunk 1]
```diff
@@ -8,43 +9,39 @@ module Check
...
+                  .downcase
+                  .include?("encoding=\"#{expected_declaration.downcase}\"")
 
-      raise(
-        WrongEncoding.new(file: filename,
-                          problem: "Missing 'encoding=\"UTF-8\"' declaration.")
-      )
+      WrongEncoding.new(file: file,
+                        problem: "Missing 'encoding=\"UTF-8\"' declaration.")
+    rescue ArgumentError
```
[the remark this answers is not in the document]
Comment (dunn) [reply lib/check/encoding.rb:33]: https://www.youtube.com/watch?v=h7975tw0EJA

**Comment by dunn** [comment]
Oh, our UTF-8 fixtures aren't pure enough for them:
```
irb(main):005:0> CharlockHolmes::EncodingDetector.detect("./test/fixtures/csv/pamss045.csv")
=> {:type=>:text, :encoding=>"ISO-8859-1", :ruby_encoding=>"ISO-8859-1", :confidence=>41, :language=>"es"}

**Comment by mcritchlow** [comment]
WTF UTF

**Comment by dunn** [comment]
Our current method of catching errors while parsing the files is actually more reliable, so let's not do this.

**Pull request closed without merging**